content/shmem_collective_wait.tex

```diff
@@ -0,0 +1,39 @@
+\apisummary{
+    The routine waits until a operation identified by a request
+    object completes.
+}
+
+\begin{apidefinition}
+
+\begin{Csynopsis}
+int @\FuncDecl{shmem\_req\_wait}@(shmem_req_h request);
+\end{Csynopsis}
+
+\begin{apiarguments}
+
+  \apiargument{IN}{request}{Request handle}
+
+\end{apiarguments}
+
+\apidescription{
+
+The \FUNC{shmem\_req\_wait} function is a blocking operation used to determine whether an
+operation identified by the request object has been completed. If the operation
+is completed, \FUNC{shmem\_req\_wait} returns zero and deallocates the request object. If
+the operation has not been completed, \FUNC{shmem\_req\_wait} blocks until the operation
+completes and then returns zero.
+
+
+In a multithreaded environment, \FUNC{shmem\_req\_wait} can be called by different
+threads but on different request objects. It is the responsibility of the
+\openshmem user to ensure that proper synchronization is used to prevent race
+conditions or deadlock. Specifically, the \FUNC{shmem\_req\_wait} operation should
+be called after the collective operation to ensure that the request object is
```

Comment:
If the requests are always deallocated in `wait` or `test` call, and if request handles are not reused, it will not run into the scenario of "premature deallocation".